scrapy/http/request/__init__.py

```diff
@@ -65,12 +65,12 @@ def _set_params(self, params):
         if params is None:
             self._params = None
             return
-        if isinstance(params, dict):
+        if any([isinstance(params, dict), isinstance(params, list)]):
```

Comment:
You can simplify this as:

```suggestion
        if isinstance(params, (dict, list)):
```

However, since urlencode supports additional types (e.g. tuples), maybe we should just pass the argument to `urlencode` and let it complain if it is not supported.

Reply:
+1. If the types do not match, the compiler throws their own error.